CDAP-13558 Support CDC pipelines on Spark version 2.x

mlozbin-cybervisiontech opened the pull request:
JIRA - https://issues.cask.co/browse/CDAP-13558
WIKI - https://wiki.cask.co/display/CE/CDC+Solution+Enhancements

In scope of this PR:
* changed widgets to follow google plugins style
* fixed plugins documentations
* updated README

mlozbin-cybervisiontech committed:
CDAP-13558 Support CDC pipelines on Spark version 2.x
- changed widgets to follow google plugins style
- fixed plugins documentations

mlozbin-cybervisiontech commented:
@albertshau, could you please take a look.

albertshau commented on docs/CDCBigTable-sparksink.md:
```diff
@@ -1,16 +1,45 @@
-CDC Google Bigtable Sink
-==========
+# CDC Google Bigtable Sink
```
Looks like the official name is "Cloud Bigtable", so let's change "Google Bigtable" to "Google Cloud Bigtable" here, and call it "Cloud Bigtable" everywhere after this

mlozbin-cybervisiontech replied:
fixed

albertshau commented on docs/CDCBigTable-sparksink.md:
```diff
@@ -1,16 +1,45 @@
-CDAP Plugin for Change Data Capture (CDC) in Google Bigtable using the Spark Framework. Plugin can be configured only for real-time pipelines.
+Description
+-----------
+CDAP Realtime Plugin for Change Data Capture (CDC) in Google Bigtable. 
```
This plugin takes input from a CDC source and writes the changes to Cloud Bigtable.

mlozbin-cybervisiontech replied:
fixed

albertshau commented on docs/CDCDatabase-streamingsource.md:
```diff
@@ -0,0 +1,30 @@
+# Change Tracking SQL Server Streaming Source
```
Change Tracking -> CDC

mlozbin-cybervisiontech replied:
fixed

albertshau commented on README.md:
```diff
@@ -9,5 +12,53 @@ to sinks such as Kudu, HBase, and Database.
+users, release announcements, and any other discussions that we think will be helpful
+to the users.
+
+## Slack Team
```
I don't think many people are monitoring the slack channel anymore, let's remove this section.

mlozbin-cybervisiontech replied:
fixed

albertshau commented on docs/CDCDatabase-streamingsource.md:
```diff
@@ -0,0 +1,30 @@
+
+Description
+-----------
+Streaming source for reading through Golden Gate Kafka topic.
```
This plugin reads Change Data Capture (CDC) events from a Golden Gate Kafka topic.

mlozbin-cybervisiontech replied:
fixed

albertshau commented on docs/CDCHBase-sparksink.md:
```diff
@@ -1,16 +1,24 @@
-CDAP Plugin for Change Data Capture (CDC) in HBase using the Spark Framework. Plugin can be configured only for real-time pipelines.
+Description
+-----------
+CDAP Realtime Plugin for Change Data Capture (CDC) in HBase.
```
This plugin takes input from a CDC source and writes the changes to HBase. It will write to the HBase instance running on the cluster.

mlozbin-cybervisiontech replied:
fixed

albertshau commented on docs/CDCKudu-sparksink.md:
```diff
@@ -0,0 +1,32 @@
+# Change Tracking SQL Server Streaming Source
```
update title

mlozbin-cybervisiontech replied:
fixed

albertshau commented on docs/CDCKudu-sparksink.md:
```diff
@@ -0,0 +1,32 @@
+
+Description
+-----------
+CDAP Realtime Plugin for Change Data Capture (CDC) in Kudu.
```
This plugin takes input from a CDC source and writes the changes to Kudu

mlozbin-cybervisiontech replied:
fixed

mlozbin-cybervisiontech committed:
CDAP-13558 Support CDC pipelines on Spark version 2.x
- introduced single message format
- added schema validation
- added config validation
- changed exception handling to follow recent guidelines
